src/osd/osd_types.h

```diff
@@ -890,6 +891,74 @@ inline ostream& operator<<(ostream& out, const pool_snap_info_t& si) {
 
 
 /*
+ * pool_opts_t
+ *
+ * pool options.
+ */
+
+class pool_opts_t {
+public:
+  enum key_t {
+    SCRUB_MIN_INTERVAL,
+    SCRUB_MAX_INTERVAL,
+    DEEP_SCRUB_INTERVAL,
+  };
+
+  enum type_t {
+    STR,
+    INT,
+    DOUBLE,
+  };
+
+  struct opt_desc_t {
+    key_t key;
+    type_t type;
+
+    opt_desc_t(key_t k, type_t t) : key(k), type(t) {}
+  };
+
+  typedef boost::variant<std::string,int,double> value_t;
```

Comment:
@trociny on most architectures, `int` is a 32bit integer. but it's not guaranteed. ceph use little endian internally. `int32_t` and its friends have their encoders defined in `encoding.h` using their corresponding little endian counterparts. but this does not apply to `int`.

actually i think the reason why https://github.com/ceph/ceph/pull/6081/files#diff-0bd9f9613a09a1111a9851762e70fcd4R979 compiles is that the `int` variable is promoted to some other type which we are able to encode.

> why other types (std::string, double) don't worry you?

good question. as to `std::string`, we do define a codec for it in `encoding.h`. but when it comes to `double`, in theory, strictly speaking, i am worried. because we don't do the byteorder on bigendian machine even we are ensure that the compiler follows IEEE754, which is not checked.